builtin/logical/database/path_rotate_credentials.go

```diff
@@ -77,6 +97,57 @@ func (b *databaseBackend) pathRotateCredentialsUpdate() framework.OperationFunc
 		return nil, nil
 	}
 }
+func (b *databaseBackend) pathRotateRoleCredentialsUpdate() framework.OperationFunc {
+	return func(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+		name := data.Get("name").(string)
+		if name == "" {
+			return logical.ErrorResponse("empty role name attribute given"), nil
+		}
+
+		role, err := b.StaticRole(ctx, req.Storage, data.Get("name").(string))
+		if err != nil {
+			return nil, err
+		}
+		if role == nil {
+			return logical.ErrorResponse("cannot rotate credentials of non-static accounts"), nil
```

Comment:
We'll be here if the role isn't found, too. Considered rewording the error to cover both cases.

Reply:
Reworded in c30b33fc7